runtime/lua/vim/lsp/handlers.lua

```diff
@@ -157,25 +157,25 @@ M['workspace/workspaceFolders'] = function(_, _, ctx)
   local client_id = ctx.client_id
   local client = vim.lsp.get_client_by_id(client_id)
   if not client then
-    err_message("LSP[id=", client_id, "] client has shut down after sending the message")
+    err_message('LSP[id=', client_id, '] client has shut down after sending the message')
     return
   end
   return client.workspace_folders or vim.NIL
 end
 
 M['textDocument/publishDiagnostics'] = function(...)
-  return require('vim.lsp.diagnostic').on_publish_diagnostics(...)
+  return require 'vim.lsp.diagnostic'.on_publish_diagnostics(...)
 end
 
 M['textDocument/codeLens'] = function(...)
-  return require('vim.lsp.codelens').on_codelens(...)
+  return require 'vim.lsp.codelens'.on_codelens(...)
```

Comment:
This is why I asked about the separate option for strings, by the way: Without the parens, it looks like you're indexing the string rather than the result of the function call...

Reply:
Sure, now that the option is available (thanks! that was fast!), this is not a problem. I just wanted to mention this as an illustration of my earlier point since I happened to come across it.

Reply:
this feature has been implement by option
```
call_arg_parentheses = unambiguous_remove_string_only
```